server/api/transport_error.go

```diff
@@ -0,0 +1,153 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+)
+
+// erroer interface is implemented by response structs to encode business logic errors
+type errorer interface {
+	error() error
+}
+
+type jsonError struct {
+	Message string              `json:"message"`
+	Errors  []map[string]string `json:"errors,omitempty"`
+}
+
+// use baseError to encode an jsonError.Errors field with an error that has
+// a generic "name" field. The frontend client always expects errors in a
+// []map[string]string format.
+func baseError(err string) []map[string]string {
+	return []map[string]string{map[string]string{
+		"name":   "base",
+		"reason": err},
+	}
+}
+
+// same as baseError, but replaces "base" with different name.
+func namedError(name string, err string) []map[string]string {
+	return []map[string]string{map[string]string{
+		"name":   name,
+		"reason": err},
+	}
+}
+
+// encode error and status header to the client
+func encodeError(ctx context.Context, err error, w http.ResponseWriter) {
```

Comment:
This looks like a direct copy/paste, which is fine, but a few of the errors are no longer necessary here, notably `osqueryError`, that's only intended for osquery endpoints.